lib/embroider/update-manifest-from-html.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const fs = require('fs');
+const findScriptSrcs = require('find-scripts-srcs-in-document');
+
+function getFileName(filePath) {
+  const scriptFileNameRegEx = /([a-zA-Z0-9_\.\-\(\):])+(\.js)/ig;
+  return Array.isArray(filePath.match(scriptFileNameRegEx)) ? filePath.match(scriptFileNameRegEx)[0] : '';
+}
+
+module.exports = function updateManifestFromHtml(root) {
+  // package.json is not present in root location skip
+  // below flow as there is no package.json to update.
+  if (!fs.existsSync(root + '/package.json')) {
+    return;
+  }
+
+  const pkg = JSON.parse(fs.readFileSync(root + '/package.json', 'UTF8'));
+  const appFile = pkg.fastboot.manifest.appFiles[0];
+
+  // if app.js file exists skip reading from the index.html as
+  // it is classic build flow
+  if(appFile && fs.existsSync(`${root}${appFile}`)) {
+    return;
+  }
+
+  // otherwise, we must parse the index.html file to figure out what's next
+  const indexHtml = fs.readFileSync(root + '/index.html', 'UTF8')
+  // TODO: fix simple-html-tokenizer: https://github.com/tildeio/simple-html-tokenizer/pull/71
+    .replace('<!DOCTYPE html>','');
+
+  const assetsDir = `assets/`;
+  const vendorFiles = pkg.fastboot.manifest.vendorFiles;
+
+  // Find script src which are not data-fastboot-ignore and data-embroider-ignore
+  const indexScriptFilePaths = findScriptSrcs(indexHtml, scriptToken => {
+    const { attributes } = scriptToken;
+    return attributes.find(attr => attr[0] === 'data-embroider-ignore' || 'data-fastboot-ignore');
+  }).filter( src => {
+    // skipping files if they are part of vendorFiles list
```

Comment:
can you briefly explain in comment, why it is safe to do so

Reply:
The Vendor Files list is generated as part of the `ember-cli-fastboot` and during the generation of pacakge.json the list of vendor files are pulled from `this.outputPaths.vendor.js` and I think that is a proper source of truth to know all the vendor files passed to fastboot env. 

As we want to skip including vendor files in the updated list of appFile, checking src with vendorFiles made more sense to me. Please let me know if i missed anything there.